### DISCUSSION TITLE
How to show a pdfplumber.display.PageImage?

### DISCUSSION THREAD

**Discussion by josephernest** [jsvine/pdfplumber#715] [Q&A] [answered]
When called from a script (and not Jupyter):

```
import pdfplumber
with pdfplumber.open("test.pdf") as pdf:
    im = pdf.pages[0].to_image()
    im.draw_rects(pdf.pages[0].extract_words())
```

how to show on screen a `pdfplumber.display.PageImage`?

This doesn't work:

```
im.show()
# or
import matplotlib.pyplot as plt
plt.show()
```

**Comment by jsvine** [comment, marked as answer]
Hi @josephernest, and great question. I've just now added that functionality, so that you can call `im.show()`: https://github.com/jsvine/pdfplumber/commit/5c7787b771c9307ae92b3c00526bf076a0cc0f8f

That's now on the `develop` branch, and will eventually be incorporated into the next release. Until then (and also afterward), you can achieve the same result with `im.annotated.show()` (of which the new method is just an alias).

Does that work for you?

**Reply by josephernest** [reply]
Wonderful! IMHO it should be added in the documentation here: https://github.com/jsvine/pdfplumber#visual-debugging
Here is a (very small!) PR: https://github.com/jsvine/pdfplumber/pull/716

**Reply by jsvine** [reply]
Thanks again for sparking this discussion, @josephernest: `im.show()` is now available in `v0.7.5`, just released.
